data/mods/Magiclysm/worldgen/elf_workshop.json

```diff
@@ -0,0 +1,102 @@
+    "furniture": {
+      "a": "f_anvil",
+      "t": "f_table",
+      "b": "f_tablesaw",
+      "c": "f_chair",
+      "d": "f_drill_press",
+      "j": "f_jointer",
+      "r": "f_router",
+      "w": "f_workbench",
+      "m": "f_workbench",
+      "s": "f_stool",
+      "f": "f_demon_forge"
+    },
+    "items": {
+      "a": [
+        { "item": "tools_anvil", "chance": 75, "repeat": [ 1, 2 ] },
+        { "item": "mithril_ingot", "chance": 10 },
+        { "item": "scrap_copper", "chance": 40 }
+      ],
+      "m": [
+        { "item": "tools_anvil", "chance": 25 },
+        { "item": "clamp", "chance": 50, "repeat": [ 1, 3 ] },
+        { "item": "metal_file", "chance": 50 },
+        { "item": "pin_reamer", "chance": 25 },
+        { "item": "mithril_ingot", "chance": 10 },
+        { "item": "scrap_copper", "chance": 80, "repeat": [ 1, 3 ] }
+      ],
+      "w": [
+        { "item": "misc_repairkit", "chance": 25 },
+        { "item": "hammer", "chance": 90 },
+        { "item": "sandpaper", "chance": 90 },
+        { "item": "clamp", "chance": 90, "repeat": [ 2, 4 ] },
+        { "item": "xacto", "chance": 10 }
+      ],
+      "f": [ { "item": "alumentum" } ]
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": "elf_workshop",
+    "object": {
+      "fill_ter": "t_metal_floor",
+      "rows": [
+        "111111111111111111111111",
+        "1fa.....fa..........jsw1",
+        "1ms......m....d....r.ct1",
+        "1d...................ct1",
+        "1bsw.........c........d1",
+        "1dsw........ctc......af1",
+        "1............c.........3",
+        "1msa...................1",
+        "1d.f..................d1",
+        "1......................3",
+        "1bsw..................b1",
+        "1....d............j....1",
+        "1...ws............sw...1",
+        "1......................1",
+        "1......................3",
+        "1......................1",
+        "1...ws............sw...1",
+        "1......................1",
+        "1...ws............sw...1",
+        "1......................1",
+        "1...........m..........1",
+        "1f.........sa.........f1",
+        "1a.sm......f.......ms.a1",
+        "111111111111111111111111"
+      ],
+      "palettes": [ "elf_palette" ],
+      "place_monsters": [
+        { "monster": "GROUP_WORKSHOP", "x": [ 3, 5 ], "y": [ 4, 23 ], "density": 0.2, "repeat": [ 1, 6 ] },
+        { "monster": "GROUP_WORKSHOP", "x": [ 3, 21 ], "y": [ 4, 23 ], "density": 0.2, "repeat": [ 1, 5 ] },
+        { "monster": "GROUP_WORKSHOP", "x": [ 3, 21 ], "y": [ 4, 23 ], "density": 0.2, "repeat": [ 1, 5 ] },
+        { "monster": "GROUP_WORKSHOP", "x": [ 3, 21 ], "y": [ 4, 23 ], "density": 0.2, "repeat": [ 1, 5 ] }
+      ]
+    }
+  },
+  {
+    "type": "overmap_special",
+    "id": "elf_workshop",
+    "overmaps": [ { "point": [ 4, 7, 3 ], "overmap": "elf_workshop_north" } ],
```

Comment:
won't this spawn at z level 3? It should probably be [ 0, 0, 0 ] here.
Also, you need to add a roof level.